### PR TITLE
Introduced holdUntil field in bundle metadata repository

### PR DESCRIPTION
**Pull request by vanruch**

### CODE REVIEW

File: src/migrations/20190521104500_add_hold_until_index.js [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+
+This Source Code Form is “Incompatible With Secondary Licenses”, as defined by the Mozilla Public License, v. 2.0.
+*/
+
+// eslint-disable-next-line import/prefer-default-export
+export const up = async (db, config, logger) => {
+  await db.collection('bundle_metadata').createIndex({'repository.holdUntil': -1});
+
+  logger.info(`Added index to bundle_metadata.repository.holdUntil`);
+};
```
Comment (Anze1m) [inline src/migrations/20190521104500_add_hold_until_index.js]: Why wouldn't we check every bundle with blockchain data and assign proper `holdUntil`?
(Also in this scenario we could determine which bundles should be erased)

File: src/services/data_model_engine.js [hunk 2]
```diff
@@ -255,7 +255,7 @@ export default class DataModelEngine {
       throw new Error(`Could not fetch the bundle from the shelterer: ${err.message || err}`);
     }
 
-    await this.bundleRepository.setBundleRepository(bundleId, BundleStatuses.downloaded, {nodeUrl});
+    await this.bundleRepository.setBundleRepository(bundleId, BundleStatuses.downloaded, {nodeUrl, holdUntil: new Date(challengeExpirationTime)});
```
Comment (Anze1m) [inline src/services/data_model_engine.js]: Why do we store `holdUntil` in two places? `metadata` and `repository`.
After successful resolution we update it only in `repository` and leave outdated in `metadata`.

File: src/services/upload_repository.js [hunk 3]
```diff
@@ -81,8 +81,8 @@ export default class UploadRepository {
     return this.shelteringWrapper.isSheltering(bundleId);
   }
 
-  async expirationDate(bundleId) {
-    return this.shelteringWrapper.shelteringExpirationDate(bundleId);
+  async expirationDateInMs(bundleId) {
```
Comment (krzysztof-jelski) [inline src/services/upload_repository.js]:
```suggestion
  async bundleExpirationDateInMs(bundleId) {
```

File: src/migrations/20190521104500_add_hold_until_index.js (hunk 1 above)
Comment (vanruch) [reply src/migrations/20190521104500_add_hold_until_index.js]: We store holdUntil in metadata already, so it shouldn't be necessary.

File: src/services/data_model_engine.js (hunk 2 above)
Comment (vanruch) [reply src/services/data_model_engine.js]: :+1: Added migration

File: src/services/upload_repository.js (hunk 3 above)
Comment (vanruch) [reply src/services/upload_repository.js]: 👍

**Pull request merged**